sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java

```diff
@@ -324,6 +329,117 @@ public FormRecognizerServiceVersion getServiceVersion() {
+     * @param target the copy authorization to the target Form Recognizer resource
+     * @param pollInterval Duration between each poll for the operation status. If none is specified, a default of
+     * 5 seconds is used.
+     *
+     * @return A {@link PollerFlux} that polls the copy model operation until it has completed, has failed,
+     * or has been cancelled.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public PollerFlux<OperationResult, CustomFormModelInfo> beginCopyModel(String modelId,
+        CopyAuthorization target, Duration pollInterval) {
+        final Duration interval = pollInterval != null ? pollInterval : DEFAULT_DURATION;
+        return new PollerFlux<OperationResult, CustomFormModelInfo>(
+            interval,
+            getCopyActivationOperation(modelId, target),
+            createCopyPollOperation(modelId),
+            (activationResponse, context) -> Mono.error(new RuntimeException("Cancellation is not supported")),
+            fetchCopyModelResultOperation(modelId));
+    }
+
+    /**
+     * Generate authorization for copying a custom model into the target Form Recognizer resource.
+     *
+     * @param resourceId Azure Resource Id of the target Form Recognizer resource where the model will be copied to.
+     * @param resourceRegion Location of the target Form Recognizer resource. A valid Azure  region name supported
+     * by Cognitive Services.
+     *
+     * <p><strong>Code sample</strong></p>
+     * {@codesnippet com.azure.ai.formrecognizer.training.FormTrainingAsyncClient.getCopyAuthorization#string-string}
+     *
+     * @return The {@link CopyAuthorization}
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<CopyAuthorization> getCopyAuthorization(String resourceId, String resourceRegion) {
+        return getCopyAuthorizationWithResponse(resourceId, resourceRegion).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Generate authorization for copying a custom model into the target Form Recognizer resource.
+     * This should be called by the target resource (where the model will be copied to) and the output can be passed as
+     * the target parameter into {@link FormTrainingAsyncClient#beginCopyModel(String, CopyAuthorization)}.
+     *
+     * @param resourceId Azure Resource Id of the target Form Recognizer resource where the model will be copied to.
+     * @param resourceRegion Location of the target Form Recognizer resource. A valid Azure  region name supported by
+     * Cognitive Services.
+     *
+     * <p><strong>Code sample</strong></p>
+     * {@codesnippet com.azure.ai.formrecognizer.training.FormTrainingAsyncClient.getCopyAuthorizationWithResponse#string-string}
+     *
+     * @return A {@link Response} containing the {@link CopyAuthorization}
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<CopyAuthorization>> getCopyAuthorizationWithResponse(String resourceId,
```

Comment:
I imagine this a Java thing, so apologies here, but why would we need in some cases the response and not others?

Reply:
The `withResponse` suffixed methods in Java are considered as the advanced scenarios where the user might be interested in getting back the `HttpResponse` and also have the ability to pass in Context (diagnostics/tracing). 
In Java, we don't want to send back the response on all methods to keep the return type simpler and easier to use.

Reply:
ohh cool. Thank you for the explanation